buffer/src/main/java/io/netty/buffer/SimpleLeakAwareCompositeByteBuf.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+
+import io.netty.util.ResourceLeakTracker;
+import io.netty.util.internal.ObjectUtil;
+
+import java.nio.ByteOrder;
+
+class SimpleLeakAwareCompositeByteBuf extends WrappedCompositeByteBuf {
+
+    final ResourceLeakTracker<ByteBuf> leak;
+
+    SimpleLeakAwareCompositeByteBuf(CompositeByteBuf wrapped, ResourceLeakTracker<ByteBuf> leak) {
+        super(wrapped);
+        this.leak = ObjectUtil.checkNotNull(leak, "leak");
+    }
+
+    @Override
+    public boolean release() {
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by unwrap().
+        ByteBuf unwrapped = unwrap();
+        if (super.release()) {
+            closeLeak(unwrapped);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean release(int decrement) {
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by unwrap().
+        ByteBuf unwrapped = unwrap();
+        if (super.release(decrement)) {
+            closeLeak(unwrapped);
+            return true;
+        }
+        return false;
+    }
+
+    private void closeLeak(ByteBuf trackedByteBuf) {
+        // Close the ResourceLeakTracker with the tracked ByteBuf as argument. This must be the same that was used when
+        // calling DefaultResourceLeak.track(...).
+        boolean closed = leak.close(trackedByteBuf);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "closed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)